xml_converter/tests/attribute_testing.py

```diff
@@ -0,0 +1,165 @@
+
+        return filtered_diff
+
+    except (FileNotFoundError, PermissionError) as e:
+        print(f"Error opening file {file_path1} or {file_path2}")
+        print(e)
+        return [str(e)]
+
+
+patterns_for_noisy_lines = [
+    "^The taco parse function took [0-9]+ milliseconds to run$",
+    "^The xml write function took [0-9]+ milliseconds to run$",
+    "^The protobuf read function took [0-9]+ milliseconds to run$",
+    "^The protobuf write function took [0-9]+ milliseconds to run$",
+    "^$"
+]
+
+
+def remove_noisy_lines(array: List[str]) -> List[str]:
+    filtered_array = []
+    for line in array:
+        match_found: bool = False
+        for pattern in patterns_for_noisy_lines:
+            if re.fullmatch(pattern, line):
+                match_found = True
+                break
+        if not match_found:
+            filtered_array.append(line)
+    return filtered_array
+
+
+def main(args: argparse.Namespace) -> None:
+    try:
+        with open(json_file_path, 'r') as json_file:
+            data = json.load(json_file)
+
+        for attribute_name in data.keys():
+            print(attribute_name)
+            attribute_data = data[attribute_name]
+
+            # Ensure that the test output directory is created
+            output_dir_path = attribute_data["path"] + "-output/"
+            if not os.path.exists(output_dir_path):
+                try:
+                    os.makedirs(output_dir_path)
+                except OSError as e:
+                    print(f"Error: {e}")
+
+            for test in attribute_data["tests"].keys():
+                file_name: str = attribute_name + test + ".xml"
+                input_xml_path = os.path.join(attribute_data["path"], file_name)
+                output_xml_path = os.path.join(output_dir_path, file_name)
+
+                result = run_xml_converter(input_xml=[input_xml_path], output_xml=[output_xml_path])
+                # Remove noisy lines
+                stdout = remove_noisy_lines(result.stdout.split("\n"))
+                stderr = remove_noisy_lines(result.stderr.split("\n"))
+                xml_diff = compare_files(input_xml_path, output_xml_path)
+
+                # Prints the results rather than comparing them to a file
+                if args.print:
+                    print(f"Test {attribute_name}{test}")
+                    print(f"'output_stdout' : {json.dumps(stdout)}")
+                    print(f"'output_stderr' = {json.dumps(stderr)}")
+                    print(f"'expected_xml_diff' = {json.dumps(xml_diff)}")
+                    continue
+
+                all_tests_passed: bool = True
+                expected_results = attribute_data["tests"][test]
+
+                if stdout != expected_results["output_stdout"]:
+                    print(f"Output did not match for test {attribute_name}{test}")
+                    print(f"Expected stdout {expected_results['output_stdout']} and got {stdout}")
```

Comment:
If a test fails lets print out the diff of the expected and the actual. Same for the others.